### ISSUE TITLE
UnicodeDecodeError while trying to run a query.

### ISSUE THREAD

**Issue by bernardko** [josiahcarlson/rom#115]
Just want to report an error I'm encountering while running a query. 

My application most likely encoded a string incorrectly and this was written to Redis. Offending string is 'D�cor'. When I try to make a query after this string exists in a model instance, it will throw this error and various queries that loads this instance fails. Unfortunately, I have to delete or fix this string before queries will work again.

```
>>> Category.query.limit(0,10).execute()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/query.py", line 670, in execute
    return list(self)
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/query.py", line 580, in _iter_all_hscan
    ent = self._model(_loading=True, **mdata)
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/model.py", line 234, in __init__
    setattr(self, attr, data)
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/columns.py", line 337, in __set__
    self._init_(obj, *value)
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/columns.py", line 577, in _init_
    value = value.decode('utf-8').encode('latin-1')
  File "/mnt/data/dev/holo/venv/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xe9 in position 1: invalid continuation byte
```

**Comment by josiahcarlson** [comment]
Is it possible that data was set directly inside Redis or via another library, instead of passing through rom? Can you show me your model definition? Are you going to be sticking with Python 2.7, or will you be moving to Python 3.x soon? Are you using Redis 2.8.12 or greater? If you answer these questions, I can probably write something that should be able to clean models where this is happening for you. If you'd like to contact me off-list, my email address is on my profile.

**Comment by bernardko** [comment]
The data was set passing through rom. Here is a simple example.

```
>>> import rom
>>> from rom import util
>>> class Category(rom.Model):
...  name = rom.String()
...
>>> category = Category(name='D�cor')
>>> category.save()
2
>>> Category.query.execute()
[<__main__.Category object at 0x7fecc62dd5d0>]
>>> Category.query.execute()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/query.py", line 670, in execute
    return list(self)
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/query.py", line 580, in _iter_all_hscan
    ent = self._model(_loading=True, **mdata)
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/model.py", line 234, in __init__
    setattr(self, attr, data)
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/columns.py", line 337, in __set__
    self._init_(obj, *value)
  File "/mnt/data/dev/holo/venv/local/lib/python2.7/site-packages/rom/columns.py", line 576, in _init_
    value = value.decode('utf-8').encode('latin-1')
UnicodeEncodeError: 'latin-1' codec can't encode character u'\ufffd' in position 1: ordinal not in range(256)
```
Maybe rom needs an extra validation which checks if the string can be encode/decode to latin-1 before it can be written to redis? Or maybe just keep the encoding in utf-8 to avoid this error? I'm not sure what is the best way to handle this.

I'm testing this on redis 3.0.6 on from apt package on Ubuntu Xenial. 

Regarding Python 2.7/3.x, my codebase is still in 2.7 but will eventually port to 3.x.

I don't need any do any cleaning on the models as I'm just evaluating using rom at the moment. Just wanted to report this issue I encountered.

Thanks!

**Comment by josiahcarlson** [comment]
Rom shouldn't even be allowing that character through, unless it's getting mangled. I've run into a few bugs related to Python storing json in Redis, and Redis json unicode being a bit funky. I wonder if this is another case of weird round-tripping bugs. I'll do some digging, thank you for the report.

**Comment by josiahcarlson** [comment]
Turns out I had been a bit too aggressive with my unicode encode/decode shenanigans. This should now be fixed in 0.40.2. If this is still an issue for you (shouldn't be), I can reopen if you've got more examples that break for you.

Thank you for the bug report!

**Comment by bernardko** [comment]
Thanks for the fix! I'll try it out again and see if this continues to happen.
